front/src/applications/editor/tools/rangeEdition/components/RangeEditionLeftPanel.tsx

```diff
@@ -75,23 +80,20 @@ const RangeEditionLeftPanel = () => {
   };
 
   const toggleSpeedRestriction = () => {
-    if (speedRestriction) resetSpeedRestrictionSelections();
-    setSpeedRestriction(!speedRestriction);
-    const selectiontype = speedRestriction ? 'idle' : 'selectSwitch';
+    if (isSpeedRestriction) resetSpeedRestrictionSelections();
+    const selectiontype = isSpeedRestriction ? 'idle' : 'selectSwitch';
```

Comment:
This function is not very clear as you setState twice if isSpeedRestriction is true, you can change it like this:
```
  const toggleSpeedRestriction = () => {
    const newEntity = cloneDeep(entity) as SpeedSectionEntity;
    newEntity.properties.track_ranges = [];
    let otherProps: Partial<RangeEditionState<SpeedSectionEntity>> = {};

    if (isSpeedRestriction) {
      newEntity.properties.on_routes = [];
      otherProps = {
        isSpeedRestriction: false,
        selectedSwitches: {},
        routeElements: {},
        highlightedRoutes: [],
        interactionState: { type: 'idle' },
      };
    } else {
      newEntity.properties.extensions = undefined;
      otherProps = { isSpeedRestriction: true, interactionState: { type: 'selectSwitch' } };
    }

    setState({
      entity: newEntity,
      ...otherProps,
    });
  }
```

Reply:
I agree, I removed the double setState. I refactoed a bit.